gcp/biganimal-csp-setup

```diff
@@ -0,0 +1,281 @@
+    echo -e "${GREEN}Run GCP CSP setup with gcloud cli ${cliversion}${NC}"
+  fi
+}
+
+check_project()
+{
+  if [[ "${project}" == "" ]]; then
+    echo -e "${RED}Error: missing -p, --project to specify the GCP project ID${NC}"
+    show_help
+    exit 1
+  fi
+  # check that the gcloud cli is configured to work with the given project and we have access to
+  set +e
+  current_project=$(gcloud config get-value project)
+  if [ $? -ne 0 ]; then
+    echo -e "${RED}Error while checking GCP credentials/profile configuration, please init/configure your gcloud CLI session/profile${NC}"
+    show_help
+    exit 1
+  fi
+  set -e
+  if [[ ! "${current_project}" == "$project" ]]; then
+    echo -e "${RED}Error: gcloud CLI not configured to work with the requested project $project!${NC}"
+    echo -e "${RED}The configured project ID is ${current_project}!${NC}"
+    show_help
+    exit 1
+  fi
+}
+
+check_names()
+{
+  if [[ "${suffix_sa_name}" =~ [_] ]]; then
+    echo -e "${RED}Error: -s, --sa-name-suffix value can't contains underscores (_)${NC}"
+    show_help
+    exit 1
+  fi
+  if [[ "${suffix_role_name}" =~ [-] ]]; then
+    echo -e "${RED}Error: -r, --role-name-suffix value can't contains dashes (-)${NC}"
+    show_help
+    exit 1
+  fi
+}
+
+while [[ $# -gt 0 ]]; do
+  key="$1"
+
+  case $key in
+    -h|--help)
+      show_help
+      exit 0
+      ;;
+    -p|--project)
+      project="$2"
+      shift 2
+      ;;
+    -s|--sa-name-suffix)
+      suffix_sa_name="$2"
+      shift 2
+      ;;
+    -r|--role-name-suffix)
+      suffix_role_name="$2"
+      shift 2
+      ;;
+    *)
+      shift
+      ;;
+  esac
+done
+
+download_dependencies()
+{
+  # a shell func to download all the dependecies of this script before starting to execute
+  # Download the latest biganimal role template or overwrite it
+  template_file=biganimal_GCP_baserole.yaml
+  curl https://raw.githubusercontent.com/EnterpriseDB/cloud-utilities/main/gcp/${template_file} -o "${CURRENT_PATH}"/${template_file}
+  if [[ -s "${CURRENT_PATH}/${template_file}" ]]; then
+    echo "Downloaded ${CURRENT_PATH}/${template_file}"
+  else
+    echo "Error: failed to download ${CURRENT_PATH}/${template_file}"
+    exit 1
+  fi
+}
+
+create_iam_custom_role()
+{
+  # verify if a custom role with the same name is already existing
+  role_exists=$(gcloud iam roles list --project "$project" \
+    --filter "name:biganimal_${suffix_role_name}" --format json | jq '. | length')
+  if [ "$role_exists" -gt 0 ]; then
+    # a custom role with the same name already exists
+    echo -e "Updating GCP custom role ${GREEN}biganimal_${suffix_role_name}${NC} within GCP project ${GREEN}${project}${NC}..."
+
+    # updating the GCP custom role
+    custom_role_id=$(gcloud iam roles list --project "$project" \
+      --filter "name:biganimal_${suffix_role_name}" --format json | jq -r '.[].name')
+
+    gcloud iam roles update "${custom_role_id##*/}" --project "$project" \
```

Comment:
With this command, we can update existing role with new permissions. For update-csp, we can reuse this script, am I right?

Reply:
yes you're right, this works. So we can re-use the same script to update the existing custom role with the new content, and I've already tested it on the dev project